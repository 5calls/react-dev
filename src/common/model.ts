
export interface Issue {
  id: string;
  name: string;
  reason: string;
  script: string;
  contacts?: Contact[];
  contactType?: string;
  categories: Category[];
  inactive: boolean;
  outcomeModels: Outcome[];
  link: string;
  linkTitle: string;
  slug: string;
}

export const DefaultIssue: Issue = {
  id: '',
  name: '',
  reason: '',
  script: '',
  categories: [],
  inactive: false,
  outcomeModels: [],
  link: '',
  linkTitle: '',
  slug: '',
};

export interface Outcome {
  label: string;
  status: string;
}

export interface Contact {
  id: string;
  name: string;
  phone: string;
  photoURL?: string;
  party: Party;
  state: string;
  reason: string;
  area?: string;
  field_offices?: FieldOffice[];
}

export interface VoterContact {
  id: string;
  name: string;
  city: string;
  phone: string;
}

export interface Category {
  name: string;
}

export class CategoryMap {
  category: Category;
  issues: Issue[];
}

export type Party = 'Democrat' | 'Republican' | 'Independent' | '';

// export const DefaultContact: Contact = {
//   name: '',
//   phone: '',
//   party: '',
//   state: '',
//   reason: ''
// };

export const DefaultContact: Contact = {} as Contact;

export interface FieldOffice {
  city: string;
  phone: string;
}

export interface UserStat {
  all: string[];
  contactedCount: number;
  voiceMailCount: number;
  unavailableCount: number;
}

export interface Group {
  id: string;
  name: string;
  description: string;
  totalCalls: number;
}

/**
 * Represents the place used to get location.
 * It may be one of three options:
 * 1. CACHED_ADDRESS - address stored in local storage
 * 2. BROWSER_GEOLOCATION - address obtained from the browser geolocation API
 * 3. IP_INFO - address obtained from ipinfo.io API
 */
export enum LocationFetchType {
  CACHED_ADDRESS = 'CACHED_ADDRESS', // 'address' in Choo version
  BROWSER_GEOLOCATION = 'BROWSER_GEOLOCATION', // 'browserGeolocation' in Choo version
  IP_INFO = 'IP_INFO' // 'ipAddress' in Choo version
}

/**
 * Encapsulates a location using latitude
 * and longitude. Undefined for either longitude
 * or latitude indicates that the geolocation has
 * has not been set.
 */
export interface GeolocationPosition {
  longitude: number | undefined;
  latitude: number | undefined;
}

/* 5 Calls API data */
export interface ApiData {
  splitDistrict: boolean;
  invalidAddress: boolean;
  normalizedLocation: string | undefined;
  divisions: string[];
  issues: Issue[];
}

export interface GroupIssues {
  splitDistrict: boolean;
  invalidAddress: boolean;
  normalizedLocation: string | undefined;
  divisions: string[];
  issues: Issue[];
}

export interface CountData {
  count: number; // total call count
}

/* Data from iponfo.io API */
export interface IpInfoData {
  ip: string;
  hostname: string;
  city: string;
  region: string; // state
  country: string;
  loc: string; // long, lat - used in issue lookup
  org: string; // internet service provider
  postal: string; // zip code
}

export enum LocationUiState {
  FETCHING_LOCATION = 'FETCHING_LOCATION',
  LOCATION_FOUND = 'LOCATION_FOUND',
  ENTERING_LOCATION = 'ENTERING_LOCATION',
  LOCATION_ERROR = 'LOCATION_ERROR'
}

export interface DonationGoal {
  goal: Donations; //
}

export interface Donations {
    count: number; // number of donors
    amount: number; // total collected (api===amount)
    total: number; // goal (api===total)
    kind: string; // denomincation (dollars)
}
