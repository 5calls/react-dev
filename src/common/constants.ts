
export const APP_NAME = '5 Calls';
export const APP_URL = 'https://5calls.org';
// export const APP_URL = 'http://localhost:8090';
export const API_URL = 'https://api.5calls.org/v1';
export const ISSUES_API_URL = `${APP_URL}/issues/?all=true&address=`;
export const REPORT_API_URL = `${APP_URL}/report`;
export const COUNTS_API_URL = `${API_URL}/counts`;
export const DONATIONS_API_URL = `${API_URL}/donations`;
export const GROUP_API_URL = `${API_URL}/groups`;
export const IP_INFO_URL = 'https://ipinfo.io/json';

export const zipCodeRegex: RegExp = /^\d{5}(-\d{4})?$/;

export const contact = {
  email: 'make5calls@gmail.com',
  github: 'https://github.com/5calls',
  twitter: 'https://twitter.com/make5calls',
  facebook: 'https://www.facebook.com/make5calls',
  apps: 'https://crgj.app.link/7R2bEB0R4F',
};
