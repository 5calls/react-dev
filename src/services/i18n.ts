import * as i18n from 'i18next';
import * as XHR from 'i18next-xhr-backend';
// import Cache from 'i18next-localstorage-cache';
import * as LanguageDetector from 'i18next-browser-languagedetector';

const instance = i18n
    .use(XHR)
    // .use(Cache)
    .use(LanguageDetector)
    .init({
        backend: {
            loadPath: '/locales/{{lng}}.json'
        },
        fallbackLng: 'en',

        react: {
            wait: true, // globally set to wait for loaded translations in translate hoc
            // exposeNamespace: true // exposes namespace on data-i18next-options to be used in eg. locize-editor
        },

        // have a common namespace used around the full app
        ns: [''],
        defaultNS: '',

        debug: true,

        // cache: {
        //   enabled: true
        // },

        interpolation: {
            escapeValue: false, // not needed for react!!
            formatSeparator: ','
        }
    });

export default instance;
