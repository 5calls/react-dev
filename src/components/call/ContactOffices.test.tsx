import * as React from 'react';
import { shallow } from 'enzyme';
import i18n from '../../services/i18n';
import { ContactOffices } from './index'
import { Issue, DefaultIssue, Contact, DefaultContact, FieldOffice } from '../../common/model';

test('Contact offices should display if available', () => {
  const fieldOffice: FieldOffice = {city: 'San Francisco', phone: '5555551212' }
  const contact: Contact = Object.assign({}, DefaultContact, { id: '101', name: 'dude', phone: '5555551212', party: '', state: 'CA', reason: 'this is your dude', fieldOffices: [fieldOffice] } )
  const issue: Issue = Object.assign({}, DefaultIssue, { id: '1', name: 'testName', contacts: [contact] });

  const component = shallow(
    <ContactOffices
      currentIssue={issue}
      contactIndex={0}
      t={i18n.t}
    />
  );
  const node = component.find('call__contact__field-office-list');
  expect(node.children.length).toEqual(1);
});

test('Contact offices should not display if unavailable', () => {
  const contact: Contact = Object.assign({}, DefaultContact, { id: '101', name: 'dude', phone: '5555551212', party: '', state: 'CA', reason: 'this is your dude' } )
  const issue: Issue = Object.assign({}, DefaultIssue, { id: '1', name: 'testName', contacts: [contact] });

  const component = shallow(
    <ContactOffices
      currentIssue={issue}
      contactIndex={0}
      t={i18n.t}
    />
  );
  const node = component.find('call__contact__field-office-list');
  expect(node).toBeUndefined;
});