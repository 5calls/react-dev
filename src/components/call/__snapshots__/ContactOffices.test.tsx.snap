// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact offices should display if available 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <span />,
  "nodes": Array [
    <span />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ContactOffices
        contactIndex={0}
        currentIssue={
                Object {
                        "categories": Array [],
                        "contacts": Array [
                          Object {
                            "fieldOffices": Array [
                              Object {
                                "city": "San Francisco",
                                "phone": "5555551212",
                              },
                            ],
                            "id": "101",
                            "name": "dude",
                            "outcomeModels": Array [
                              Object {
                                "label": "vm",
                                "status": "vm",
                              },
                            ],
                            "party": "",
                            "phone": "5555551212",
                            "reason": "this is your dude",
                            "state": "CA",
                          },
                        ],
                        "id": "1",
                        "inactive": false,
                        "link": "",
                        "linkTitle": "",
                        "name": "testName",
                        "outcomeModels": Array [],
                        "reason": "",
                        "script": "",
                      }
        }
        t={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ContactOffices {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "contactIndex": 0,
          "currentIssue": Object {
            "categories": Array [],
            "contacts": Array [
              Object {
                "fieldOffices": Array [
                  Object {
                    "city": "San Francisco",
                    "phone": "5555551212",
                  },
                ],
                "id": "101",
                "name": "dude",
                "outcomeModels": Array [
                  Object {
                    "label": "vm",
                    "status": "vm",
                  },
                ],
                "party": "",
                "phone": "5555551212",
                "reason": "this is your dude",
                "state": "CA",
              },
            ],
            "id": "1",
            "inactive": false,
            "link": "",
            "linkTitle": "",
            "name": "testName",
            "outcomeModels": Array [],
            "reason": "",
            "script": "",
          },
          "t": [Function],
        },
        "refs": Object {},
        "showField": [Function],
        "state": Object {
          "showFieldOfficeNumbers": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <span />,
        "_debugID": 2,
        "_renderedOutput": <span />,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ContactOffices
    contactIndex={0}
    currentIssue={
        Object {
            "categories": Array [],
            "contacts": Array [
              Object {
                "fieldOffices": Array [
                  Object {
                    "city": "San Francisco",
                    "phone": "5555551212",
                  },
                ],
                "id": "101",
                "name": "dude",
                "outcomeModels": Array [
                  Object {
                    "label": "vm",
                    "status": "vm",
                  },
                ],
                "party": "",
                "phone": "5555551212",
                "reason": "this is your dude",
                "state": "CA",
              },
            ],
            "id": "1",
            "inactive": false,
            "link": "",
            "linkTitle": "",
            "name": "testName",
            "outcomeModels": Array [],
            "reason": "",
            "script": "",
          }
    }
    t={[Function]}
/>,
}
`;

exports[`Contact offices should not display if unavailable 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <span />,
  "nodes": Array [
    <span />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ContactOffices
        contactIndex={0}
        currentIssue={
                Object {
                        "categories": Array [],
                        "contacts": Array [
                          Object {
                            "id": "101",
                            "name": "dude",
                            "party": "",
                            "phone": "5555551212",
                            "reason": "this is your dude",
                            "state": "CA",
                          },
                        ],
                        "id": "1",
                        "inactive": false,
                        "link": "",
                        "linkTitle": "",
                        "name": "testName",
                        "outcomeModels": Array [
                          Object {
                            "label": "unavailable",
                            "status": "unavailable",
                          },
                          Object {
                            "label": "contact",
                            "status": "contact",
                          },
                        ],
                        "reason": "",
                        "script": "",
                      }
        }
        t={[Function]}
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ContactOffices {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "contactIndex": 0,
          "currentIssue": Object {
            "categories": Array [],
            "contacts": Array [
              Object {
                "id": "101",
                "name": "dude",
                "party": "",
                "phone": "5555551212",
                "reason": "this is your dude",
                "state": "CA",
              },
            ],
            "id": "1",
            "inactive": false,
            "link": "",
            "linkTitle": "",
            "name": "testName",
            "outcomeModels": Array [
              Object {
                "label": "unavailable",
                "status": "unavailable",
              },
              Object {
                "label": "contact",
                "status": "contact",
              },
            ],
            "reason": "",
            "script": "",
          },
          "t": [Function],
        },
        "refs": Object {},
        "showField": [Function],
        "state": Object {
          "showFieldOfficeNumbers": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <span />,
        "_debugID": 4,
        "_renderedOutput": <span />,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ContactOffices
    contactIndex={0}
    currentIssue={
        Object {
            "categories": Array [],
            "contacts": Array [
              Object {
                "id": "101",
                "name": "dude",
                "party": "",
                "phone": "5555551212",
                "reason": "this is your dude",
                "state": "CA",
              },
            ],
            "id": "1",
            "inactive": false,
            "link": "",
            "linkTitle": "",
            "name": "testName",
            "outcomeModels": Array [
              Object {
                "label": "unavailable",
                "status": "unavailable",
              },
              Object {
                "label": "contact",
                "status": "contact",
              },
            ],
            "reason": "",
            "script": "",
          }
    }
    t={[Function]}
/>,
}
`;
